openmetadata-service/src/main/java/org/openmetadata/service/events/subscription/gchat/GChatPublisher.java

```diff
@@ -56,26 +77,19 @@ protected void onShutdownDelegate() {
   protected void sendAlert(EventResource.EventList list) {
 
     for (ChangeEvent event : list.getData()) {
-      long attemptTime = System.currentTimeMillis();
       try {
         GChatMessage gchatMessage = ChangeEventParser.buildGChatMessage(event);
-        Response response =
-            target.post(javax.ws.rs.client.Entity.entity(gchatMessage, MediaType.APPLICATION_JSON_TYPE));
-        if (response.getStatus() >= 300 && response.getStatus() < 400) {
-          // 3xx response/redirection is not allowed for callback. Set the webhook state as in error
-          setErrorStatus(attemptTime, response.getStatus(), response.getStatusInfo().getReasonPhrase());
-        } else if (response.getStatus() >= 400 && response.getStatus() < 600) {
-          // 4xx, 5xx response retry delivering events after timeout
-          setNextBackOff();
-          setAwaitingRetry(attemptTime, response.getStatus(), response.getStatusInfo().getReasonPhrase());
-          Thread.sleep(currentBackoffTime);
-        } else if (response.getStatus() == 200) {
-          setSuccessStatus(System.currentTimeMillis());
+        List<Invocation.Builder> targets = getTargetsForWebhook(webhook, G_CHAT_WEBHOOK, client, daoCollection, event);
+        if (target != null) {
+          targets.add(target);
+        }
+        for (Invocation.Builder actionTarget : targets) {
+          postWebhookMessage(this, actionTarget, gchatMessage);
         }
       } catch (Exception e) {
-        LOG.error("Failed to publish event {} to gchat due to {} ", event, e.getMessage());
+        LOG.error("Failed to publish event {} to gChat due to {} ", event, e.getMessage());
         throw new EventPublisherException(
-            String.format("Failed to publish event %s to gchat due to %s ", event, e.getMessage()));
+            String.format("Failed to publish event %s to gChat due to %s ", event, e.getMessage()));
       }
```

Comment:
Please move them to CatalogExceptionMessage as a method. This will help us localize these in the future.